File: Java/src/onreadyfyMe/Auto.java
package onreadyfyMe;

import java.text.DecimalFormat;

public class Auto extends Vehiculo {
	private String puertas;

	public String getPuertas() {
		return puertas;
	}

	public void setPuertas(String puertas) {
		this.puertas = puertas;
	}

	public Auto(String marca, String modelo, double precio, String puertas) {
		super(marca, modelo, precio);
		this.puertas = puertas;
	}

	public Auto() {
	}

	public void mostrarTodosDatos() {
		DecimalFormat formatea = new DecimalFormat("###,###.00");
		System.out.println("Marca: " + getMarca() + " // Modelo: " + getModelo() + " // Puertas: " + getPuertas()
				+ " // Precio: $" + formatea.format(getPrecio()));
	}

}

File: Python/auto.py
from vehiculo import Vehiculo

class Auto(Vehiculo):
    __puertas = 0

    def __init__(self, marca, modelo, precio, puertas):
        Vehiculo.__init__(self, marca, modelo, precio)
        self.__puertas = puertas

    def get_puertas(self):
        return self.__puertas

    def set_puertas(self, puertas):
        self.__puertas = puertas

    def __str__(self):
        return f"Marca: {self.get_marca()} // Modelo: {self.get_modelo()} // Puertas: {self.get_puertas()} // Precio: ${'{:,.2f}'.format(self.get_precio()).replace(',','~').replace('.',',').replace('~','.')}"

    def __gt__(self, auto):
        return (self.get_precio() > auto.get_precio())